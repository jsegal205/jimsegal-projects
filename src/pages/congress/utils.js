export const chamberTitles = (chamber) => {
  switch (chamber) {
    case "house":
      return "Representatives";
    case "senate":
      return "Senators";
    default:
      return "Members";
  }
};

export const individualChamberTitle = (chamber) => {
  const pluralTitle = chamberTitles(chamber);
  return pluralTitle.substring(0, pluralTitle.length - 1);
};

export const otherChamber = (chamber) =>
  chamber === "house" ? "senate" : "house";
