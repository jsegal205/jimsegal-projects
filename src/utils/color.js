const getRGB = (hex) => {
  return {
    r: parseInt(hex.slice(0, 2), 16),
    g: parseInt(hex.slice(2, 4), 16),
    b: parseInt(hex.slice(4, 6), 16),
  };
};

export const randomHexColor = () =>
  Math.floor(Math.random() * 16777215).toString(16);

export const complementaryColor = (hex) => {
  const padZero = (str) => ("00" + str).slice(-2);
  const invertColor = (digit) => padZero((255 - digit).toString(16));
  const { r, g, b } = getRGB(hex);

  return invertColor(r) + invertColor(g) + invertColor(b);
};

export const textColor = (hex) => {
  const { r, g, b } = getRGB(hex);
  const brightness = Math.round((r * 299 + g * 587 + b * 114) / 1000);

  return brightness > 125 ? "444444" : "efefef";
};
