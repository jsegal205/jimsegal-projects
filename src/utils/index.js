// export const apiUrlBase = "https://api.jimsegal.com";
export const apiUrlBase = "http://localhost:8001";

export const projectInfo = [
  {
    absolute: true,
    link: "https://jimsegal.com",
    subtitle: "OG site with other projects",
    title: "JimSegal.com",
  },
  {
    link: "/congress",
    subtitle: "Some fun data surrounding the current congressional sesssion",
    title: "Congress",
  },
  {
    link: "/isJimWearingShorts",
    subtitle: "A site to predict if Jim is wearing shorts or not",
    title: "Is Jim Wearing Shorts?",
  },
  {
    link: "mountaingoat",
    subtitle: "Digital game",
    title: "Mountain Goat Game",
  },
  {
    link: "/recipes",
    subtitle: "A bunch of recipes that I frequent and wanted to show off",
    title: "Recipes",
  },
  {
    link: "/spacex",
    subtitle: "A countdown timer until the next launch of a Space X rocket",
    title: "Space X",
  },
];
