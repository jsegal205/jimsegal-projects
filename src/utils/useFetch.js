import { useState, useEffect } from "react";
import axios from "axios";

// custom hook for performing GET request
const useFetch = (url, initialValue) => {
  const [data, setData] = useState(initialValue);
  const [loading, setLoading] = useState(true);
  useEffect(() => {
    const fetchData = async () => {
      try {
        setLoading(true);
        const response = await axios.get(url);
        setData(response.data);
      } catch ({ response }) {
        setData({ status: response.status });
      } finally {
        setLoading(false);
      }
    };
    fetchData();
  }, [url]);
  return { loading, data };
};

export default useFetch;
