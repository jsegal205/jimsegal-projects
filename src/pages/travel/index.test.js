import React from "react";
import "@testing-library/jest-dom";
import { BrowserRouter as Router } from "react-router-dom";

import { render } from "@testing-library/react";

import useFetch from "../../utils/useFetch";
import { useJsApiLoader } from "@react-google-maps/api";
import { apiUrlBase } from "../../utils";

import Travel from "./index";

jest.mock("../../utils/useFetch", () => jest.fn());
jest.mock("@react-google-maps/api", () => ({
  GoogleMap: () => {
    const React = require("react");
    class MockGoogleMap extends React.Component {
      render() {
        return <>mockGoogleMap</>;
      }
    }

    return <MockGoogleMap />;
  },

  useJsApiLoader: jest.fn(() => ({
    isLoaded: true,
  })),
}));

describe("Travel Component", () => {
  const baseUrl = `${apiUrlBase}/travel`;

  describe("when `useFetch` is awaiting promise to resolve", () => {
    it("displays `Loading...`", () => {
      useFetch.mockReturnValue({
        loading: true,
        data: [],
      });

      const { container } = render(
        <Router>
          <Travel />
        </Router>
      );
      expect(useFetch).toHaveBeenCalledWith(baseUrl);
      expect(container).toMatchSnapshot();
    });
  });

  describe("when `useFetch` returns data", () => {
    it("displays data", () => {
      useFetch.mockReturnValue({
        loading: false,
        data: [
          {
            city: "Austin",
            state: "TX",
            country: "USA",
            lat: 30.267153,
            lng: -97.743061,
            visits: ["February 2018", "April 2014"],
          },
        ],
      });

      useJsApiLoader.mockReturnValue({
        isLoaded: true,
      });

      const { container } = render(
        <Router>
          <Travel />
        </Router>
      );
      expect(useFetch).toHaveBeenCalledWith(baseUrl);
      expect(container).toMatchSnapshot();
    });

    describe("when 500 status returned", () => {
      it("displays error messaging", () => {
        useFetch.mockReturnValue({
          loading: false,
          data: {
            error: "server likely on fire",
            status: 500,
          },
        });

        const { container } = render(
          <Router>
            <Travel />
          </Router>
        );
        expect(useFetch).toHaveBeenCalledWith(baseUrl);
        expect(container).toMatchSnapshot();
      });
    });
  });
});
