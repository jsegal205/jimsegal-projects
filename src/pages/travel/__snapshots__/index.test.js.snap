// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Travel Component when \`useFetch\` is awaiting promise to resolve displays \`Loading...\` 1`] = `
<div>
  <section>
    <h2>
      My Travels
    </h2>
    <small>
      Vacation destinations since 2008
    </small>
    <section
      class="loader"
    >
      <div
        class="loader-wheel"
      />
      <div
        class="loader-text"
      >
        Loading
      </div>
    </section>
  </section>
</div>
`;

exports[`Travel Component when \`useFetch\` returns data displays data 1`] = `
<div>
  <section>
    <h2>
      My Travels
    </h2>
    <small>
      Vacation destinations since 2008
    </small>
    mockGoogleMap
    <small>
      Click the pins for more information
    </small>
  </section>
</div>
`;

exports[`Travel Component when \`useFetch\` returns data when 500 status returned displays error messaging 1`] = `
<div>
  <section>
    <h2>
      My Travels
    </h2>
    <small>
      Vacation destinations since 2008
    </small>
    <section>
      <h3>
        Whoops! There was a problem loading
         
        Travel data.
      </h3>
      <p>
        Please reload browser to try again in a little bit.
      </p>
    </section>
  </section>
</div>
`;
